src/metabase/lib/dispatch.cljc

```diff
@@ -21,10 +21,5 @@
   ;; of [[toucan2.core/model]]?
   (or (mbql-clause-type x)
       (when (map? x)
-        (if (and (or (keyword? (:operator x))
-                     (string? (:operator x)))
-                 (vector? (:args x))
-                 (map? (:options x {})))
-          :lib/external-op
-          (:lib/type x)))
+        (:lib/type x))
```

Comment:
Eliminated need to special-case the dispatch logic for external op representations